src/services/fetchProductsFunction.js

```diff
@@ -0,0 +1,12 @@
+const API_BASE_URL = import.meta.env.VITE_BACKEND_URL;
+
+const fetchProductsFunction = (offset, setOffset, filteredCategory, setFilteredCategory, filter) => {
```

Comment:
it wasn't the greatest idea to make this function external but still pretty much specific for a certain component. there is almost no way of reusing it with multiple components, more over, it requires a lot of function's arguments, which is apparently bad practice.

Reply:
fetchProductsFunction and fetchCarouselItemsFunction functions is called from several components. And maybe it can be used when the product search feature by category will be implement

Reply:
👍 
please rename `fetchProductsFunction` into `fetchProducts`
or, `fetchWhateverItemsFunction` into just `fetchWhatever`